From 0a801c55500becc72ad086c0314afdebcc9b18fa Mon Sep 17 00:00:00 2001
From: Stefan Chulski <stefanc@marvell.com>
Date: Wed, 2 Jun 2021 01:06:41 -0700
Subject: [PATCH 1595/1921] net: mvpp2: Move all DSA related code under NET_DSA
 config

dsa_notifier code would be compiled only if CONFIG_NET_DSA enabled.

Change-Id: I7580208addede90468a2c2a5700531f99c1b7bc6
Signed-off-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/53257
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/mvpp2/mvpp2.h      | 2 ++
 drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c | 7 ++++++-
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/mvpp2/mvpp2.h b/drivers/net/ethernet/marvell/mvpp2/mvpp2.h
index 098546d5130c..23e559218af1 100644
--- a/drivers/net/ethernet/marvell/mvpp2/mvpp2.h
+++ b/drivers/net/ethernet/marvell/mvpp2/mvpp2.h
@@ -1083,8 +1083,10 @@ struct mvpp2_port {
 	/* Indication, whether port is connected to XLG MAC */
 	bool has_xlg_mac;
 
+#if IS_ENABLED(CONFIG_NET_DSA)
 	/* Notifier required when the port is connected to the switch */
 	struct notifier_block dsa_notifier;
+#endif
 
 	struct mvpp2_dbgfs_port_flow_entry *dbgfs_port_flow_entry;
 };
diff --git a/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c b/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
index 709ce85b7c24..d33dfbb9ae6e 100644
--- a/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
+++ b/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
@@ -6585,6 +6585,7 @@ static const struct phylink_mac_ops mvpp2_phylink_ops = {
 	.mac_link_down = mvpp2_mac_link_down,
 };
 
+#if IS_ENABLED(CONFIG_NET_DSA)
 /* DSA notifier */
 static void mvpp2_dsa_port_register(struct net_device *dev)
 {
@@ -6612,6 +6613,7 @@ static int mvpp2_dsa_notifier(struct notifier_block *unused,
 
 	return NOTIFY_DONE;
 }
+#endif
 
 /* Ports initialization */
 static int mvpp2_port_probe(struct platform_device *pdev,
@@ -6896,6 +6898,7 @@ static int mvpp2_port_probe(struct platform_device *pdev,
 		spin_lock_init(&port->tx_lock[i]);
 	}
 
+#if IS_ENABLED(CONFIG_NET_DSA)
 	/* Register DSA notifier */
 	port->dsa_notifier.notifier_call = mvpp2_dsa_notifier;
 	err = register_dsa_notifier(&port->dsa_notifier);
@@ -6903,7 +6906,7 @@ static int mvpp2_port_probe(struct platform_device *pdev,
 		dev_err(&pdev->dev, "failed to register DSA notifier\n");
 		goto err_phylink;
 	}
-
+#endif
 	return 0;
 
 err_phylink:
@@ -6934,7 +6937,9 @@ static void mvpp2_port_remove(struct mvpp2_port *port)
 	mvpp2_port_musdk_set(port->dev, false);
 	kfree(port->dbgfs_port_flow_entry);
 	unregister_netdev(port->dev);
+#if IS_ENABLED(CONFIG_NET_DSA)
 	unregister_dsa_notifier(&port->dsa_notifier);
+#endif
 	if (port->phylink)
 		phylink_destroy(port->phylink);
 	free_percpu(port->pcpu);
-- 
2.31.1

